code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -334,6 +334,10 @@
 		for(var/i in loot)
 			new i(loc)
 
+/mob/living/simple_animal/revive()
+	..()
+	density = initial(density)

```

Comment:
I'd still rather figure out why density is breaking.

Reply:
It's because density is set to 0 on death(), so people can walk over the corpse, I assume. I just reset the thing that previously wasn't reset after revival.

Reply:
I don't know of a better solution than this at the moment, it is what TG did to fix the issue.